### PR TITLE
Review program:object-detection-armnn-tflite

### PR DESCRIPTION
**Pull request by psyhtest**
Let's review `program:object-detection-armnn-tflite` together, and make similar changes to `program:object-detection-tflite` too.

### CODE REVIEW

File: program/object-detection-armnn-tflite/includes/settings.h [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+inline std::string alter_str(std::string a, std::string b) { return a != "" ? a: b; };
+inline std::string alter_str(char *a, std::string b) { return a != nullptr ? a: b; };
+
+
+class Settings {
+public:
+    Settings() {
+
+        // Load settings
+        std::ifstream settings_file("env.ini");
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/includes/settings.h]: We should get rid of `env.ini` cause it'll be hard to make it work on Android.

File: program/object-detection-armnn-tflite/includes/non_max_suppression.h [hunk 2]
```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2018 cTuning foundation.
+ * See CK COPYRIGHT.txt for copyright details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause.
+ * See CK LICENSE.txt for licensing details.
+ */
+
+
+#ifndef NMS_NON_MAX_SUPPRESSION_H
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/includes/non_max_suppression.h]: `NMS_NON_MAX_SUPPRESSION_H` -> `NON_MAX_SUPRESSION_H`

File: program/object-detection-armnn-tflite/includes/non_max_suppression.h [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+inline void swap_float(float &a, float &b) { float c = a; a = b; b = c;}
+
+inline void swap_float(float *a, float *b) { float c = *a; *a = *b; *b = c;}
+
+inline void swap_int(int &a, int &b) { int c = a; a = b; b = c;}
+
+inline void swap_int(int *a, int *b) { int c = *a; *a = *b; *b = c;}
+
+
+// Check if `box` intersects (x1, y1, x2, y2) greater than `treshold` of area
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/includes/non_max_suppression.h]: threshold

File: program/object-detection-armnn-tflite/includes/non_max_suppression.h [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+}
+
+void postprocess_detections(Settings *s,
+                            std::vector<DetectionBox> &detection_boxes,
+                            const float *in_boxes,
+                            const float *in_scores,
+                            int src_width,
+                            int src_height,
+                            bool correct_background) {
+
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/includes/non_max_suppression.h]: Remove extra blank lines?

File: program/object-detection-armnn-tflite/includes/settings.h [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+        }
+
+        std::string nms_type = alter_str(getenv("USE_NMS"), "no");
+        if (str_to_lower(nms_type) == "no") {
+            _graph_file = std::string(getenv("CK_ENV_TENSORFLOW_MODEL_TFLITE_GRAPH_NO_NMS"));
+        } else {
+            std::cout << std::endl << "ERROR: Unsupported USE_NMS type - " << nms_type << std::endl;
+            exit(-1);
+        }
+        _graph_file = std::string(getenv("CK_ENV_TENSORFLOW_MODEL_ROOT")) + "/" + _graph_file;
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/includes/settings.h]: `/` is *nix-specific delimiter. Make more portable?

File: program/object-detection-armnn-tflite/.cm/meta.json [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+      "sort": 20,
+      "tags": "lib,tensorflow-lite,tflite,vstatic"
+    },
+    "lib-armnn": {
+      "local": "yes",
+      "name": "ArmNN library (with TFLite support)",
+      "sort": 40,
+      "tags": "lib,armnn,tflite,rel.19.02",
+      "update_tags_if_env" : {
+        "neon":         [ { "USE_NEON":   ["on", "ON", "yes", "YES", "1"] } ],
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/.cm/meta.json]: add `"true"`, `"TRUE"`?

File: program/object-detection-armnn-tflite/.cm/meta.json [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        "run_output_files": [
+          "tmp-ck-timer.json"
+        ]
+      }
+    }
+  },
+  "run_deps": {
+    "dataset": {
+      "local": "yes",
+      "name": "Object detection dataset",
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/.cm/meta.json]: Object Detection dataset

File: program/object-detection-armnn-tflite/.cm/meta.json [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+      "sort": 20,
+      "tags": "dataset,object-detection"
+    },
+    "flatbuffers": {
+      "local": "yes",
+      "name": "flatbuffers library",
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/.cm/meta.json]: FlatBuffers library

File: program/object-detection-armnn-tflite/.cm/meta.json [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  },
+  "run_deps": {
+    "dataset": {
+      "local": "yes",
+      "name": "Object detection dataset",
+      "sort": 20,
+      "tags": "dataset,object-detection"
+    },
+    "flatbuffers": {
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/.cm/meta.json]: lib-flatbuffers

File: program/object-detection-armnn-tflite/.cm/meta.json [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+    "CK_SKIP_IMAGES": 0,
+    "USE_NEON": "NO",
+    "USE_OPENCL": "NO"
+  },
+  "skip_bin_ext": "yes",
+  "source_files": [
+    "detect.cpp"
+  ],
+  "tags": [
+    "tensorflow-lite-detection",
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/.cm/meta.json]: split into several tags: "tensorflow-lite", "object-detection"

File: program/object-detection-armnn-tflite/includes/detect.hpp [hunk 11]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2018 cTuning foundation.
+ * See CK COPYRIGHT.txt for copyright details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause.
+ * See CK LICENSE.txt for licensing details.
+ */
+
+#ifndef DETECT_HEADER_FILE
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/includes/detect.hpp]: DETECT_H

File: program/object-detection-armnn-tflite/includes/settings.h [hunk 12]
```diff
@@ -0,0 +1,365 @@
...
+            std::cout << "Use Neon: " << _use_neon << std::endl;
+            std::cout << "Use OpenCL: " << _use_opencl << std::endl;
+        }
+
+        // Create results dir if none
+        auto dir = opendir(_detections_out_dir.c_str());
+        if (dir)
+            closedir(dir);
+        else
+            system(("mkdir " + _detections_out_dir).c_str());
```
Comment (psyhtest) [inline program/object-detection-armnn-tflite/includes/settings.h]: Use a more portable construct to create dir?

File: program/object-detection-armnn-tflite/includes/detect.hpp (hunk 11 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/includes/detect.hpp]: Fixed

File: program/object-detection-armnn-tflite/.cm/meta.json (hunk 10 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/.cm/meta.json]: Fixed

File: program/object-detection-armnn-tflite/.cm/meta.json (hunk 9 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/.cm/meta.json]: Fixed

File: program/object-detection-armnn-tflite/.cm/meta.json (hunk 8 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/.cm/meta.json]: Fixed

File: program/object-detection-armnn-tflite/.cm/meta.json (hunk 7 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/.cm/meta.json]: Fixed

File: program/object-detection-armnn-tflite/.cm/meta.json (hunk 6 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/.cm/meta.json]: Fixed

File: program/object-detection-armnn-tflite/includes/non_max_suppression.h (hunk 4 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/includes/non_max_suppression.h]: Fixed

File: program/object-detection-armnn-tflite/includes/non_max_suppression.h (hunk 3 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/includes/non_max_suppression.h]: Fixed

File: program/object-detection-armnn-tflite/includes/non_max_suppression.h (hunk 2 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/includes/non_max_suppression.h]: Fixed

File: program/object-detection-armnn-tflite/includes/settings.h (hunk 5 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/includes/settings.h]: Fixed

File: program/object-detection-armnn-tflite/includes/settings.h (hunk 12 above)
Comment (bellycat77) [reply program/object-detection-armnn-tflite/includes/settings.h]: Fixed

**Pull request merged**